File: Algorithms/p357_Count_Numbers_with_Unique_Digits.py
class Solution(object):
    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        res = 1
        f = 1
        for i in range(0,min(n,10)):
            res += 9*f
            f *= 9-i
        return res

File: Algorithms/p357_Count_Numbers_with_Unique_Digits/p357_Count_Numbers_with_Unique_Digits.cpp
class Solution {
public:
    int countNumbersWithUniqueDigits(int n) {
        int res = 1;
        for(int i=0,f=1;i<n && i<10;i++){
            res += 9*f;
            f *= 9-i;
        }
        return res;
    }
};
